### PR TITLE
Lit 2.0 Break Changes (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by ChrisLabattD2L**
This PR makes the breaking changes required to update to Lit 2 (`lit-html` 2 and `lit-element` 3). The Lit 2 upgrade will be a coordinated effort, so **this PR should not merge at this time**.
Breaking change checklist: 
- [x] Update `lit-html` -> `^2`, `lit-element` -> `^3`, `open-wc/testing` -> `^3` (if used)
- [x] `createRenderRoot` usages -> move from LitElement to ReactiveElement 
- [x] Remove Dependabot Lit rules
- [x] Bump any dependencies to their Lit 2 versions
- [x] Remove any manual lifecycle calls to Lit 1 reactive controllers
- [x] Backwards-compatible changes merged: https://github.com/BrightspaceUILabs/edit-in-place/pull/118
Renamed API's:
- [x] `UpdatingElement` -> `ReactiveElement`
- [x] `@internalProperty` -> `@state` (we don't use TypeScript decorators)
- [x] `static getStyles()` -> `static finalizeStyles(styles)`
- [x] `_getUpdateComplete()` -> `getUpdateComplete()`
- [x] `NodePart` -> `ChildPart`
Testing checklist: 
- [x] CI/ unit tests pass
- [x] Local testing on demo pages
- [x] Testing component in the LMS
For more information on the Lit 2 upgrade, visit https://lit.dev/docs/releases/upgrade/ or contact team Gaudi.

### CODE REVIEW

**Comment by ChrisLabattD2L** [comment]
I've tested the `edit-in-place` component in both it's own demo page and in the demo for [d2l-activity-collection-editor](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-collection-editor/d2l-activity-collection-editor.js#L510), since that's the only file that seems to call it. I haven't been able to test in the BSI yet since I can't seem to find where that page actually connects to the LMS. If that testing is enough, then I'll push the changes regarding bumping dependencies that also use Lit. since that's the last piece missing (and that's the change that will cause the tests to fail)

**Comment by annabelsegalld2l** [comment]
> 

Hi Chris, sorry I just saw this PR now. I think that testing should be enough, I wouldn't worry about BSI if you have tested the component that calls it! You can push the commit that bumps the other lit repos (which will fail for now until those new versions are actually released)

File: package.json [hunk 1]
```diff
@@ -37,8 +37,9 @@
     "eslint-plugin-sort-class-members": "^1"
   },
   "dependencies": {
-    "@brightspace-ui/core": "^1",
-    "lit-element": "^2"
+    "@brightspace-ui/core": "^2",
+    "lit-element": "^3",
+    "lit-html": "^2"
```
Comment (tenbradley) [inline package.json:42]: Do we need to add lit-html as a dependency?

Comment (annabelsegalld2l) [reply package.json:42]: It's used here: https://github.com/BrightspaceUILabs/edit-in-place/blob/master/d2l-labs-edit-in-place.js#L11. Looks like it was just missing from the dependency list (as lit-element imports lit-html, so you _technically_ don't need it), so we should add it here.

**Review by BhamelinD2L — APPROVED** [review]

**Pull request merged**